### PR TITLE
Workload ID: Restructure docs to be more value-first

### PR DESCRIPTION
**Pull request by strideynet**
Since we don't have a general landing page for Workload ID on the main marketing site, I figured it made some sense to try and make the Workload Identity introduction on the docs focus a little more on the potential use-cases and value - rather than diving straight into the technical deep-end.

### CODE REVIEW

File: docs/pages/enroll-resources/workload-identity/introduction.mdx [hunk 1]
```diff
@@ -1,157 +1,106 @@
...
+here's some of the key ones:
+
+- Eliminate the use of long-lived shared secrets within your infrastructure, and
+  reduce the risk of exfiltration and time spent by engineers creating and 
+  rotating these secrets.
+- Establish an out of the box universal form of identity for your workloads,
+  enabling your engineers to get on with building new services without needing 
+  to think about how they'll authenticate.
+- Converge on a first-class form of identity for your workloads, simplifying
+  infrastructure by reducing the number of different ways workloads authenticate.
```
Comment (ptgott) [inline docs/pages/enroll-resources/workload-identity/introduction.mdx:38]: I would shorten this introductory section by combining the list of benefits with the list of use cases.

File: docs/pages/enroll-resources/workload-identity/spiffe.mdx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+description: Learn about Secure Production Identity Framework For Everyone (SPIFFE) and how it is implemented by Teleport Workload Identity
+---
+
+SPIFFE (Secure Production Identity Framework For Everyone) is a set of standards
+for securely identifying workloads.
+
+SPIFFE sets out:
+
+- A format for uniquely specifying an identity, the SPIFFE ID.
+- Standards for encoding the SPIFFE ID into verifiable documents, the SVID.
```
Comment (ptgott) [inline docs/pages/enroll-resources/workload-identity/spiffe.mdx:12]: What does "the SVID" modify here? Is it a rephrasing of "standards"?

File: docs/pages/enroll-resources/workload-identity/spiffe.mdx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+- Processes that workloads should use to validate a received SVID.
+- A set of APIs that workloads can use to request SVIDS, the Workload API.
+
+The open nature and popularity of SPIFFE make it a great choice as a foundation
```
Comment (ptgott) [inline docs/pages/enroll-resources/workload-identity/spiffe.mdx:16]: Is there a way to use a more objective-sounding word than "great" here?

File: docs/pages/enroll-resources/workload-identity/spiffe.mdx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+for a full workload identity implementation. It is supported as an identity
+provider by a number of popular tools (such as Linkerd and Istio) and
+off-the-shelf SDKs exist for implementing SPIFFE directly into your own
+services.
+
+It's important to recognize that SPIFFE does not specify how to use SPIFFE IDs
+for authorization. This gives a high level of flexibility, allowing you to
+implement authorization in a way that suits you.
+
+## SPIFFE IDs and Trust Domains
```
Comment (ptgott) [inline docs/pages/enroll-resources/workload-identity/spiffe.mdx:26]:
```suggestion
## SPIFFE IDs and trust domains
```
We are standardizing on [sentence case in headings](https://github.com/gravitational/docs/blob/main/docs-contributors/style-guide.md#headings).

**Review by ptgott — APPROVED** [review]

File: docs/pages/enroll-resources/workload-identity/spiffe.mdx (hunk 2 above)
Comment (strideynet) [reply docs/pages/enroll-resources/workload-identity/spiffe.mdx:12]: Ah I'll flip these definitions around to make them more clearly definitions.

**Review by ryanclark — APPROVED** [review]

**Pull request merged**